capstone/src/test/java/com/google/univiz/api/resource/MapsResourceImplTest.java

```diff
@@ -0,0 +1,151 @@
+    List<CollegeId> ids = Lists.newArrayList(id);
+    assertThat(ids).hasSize(1);
+
+    CollegeData college =
+        CollegeData.builder()
+            .setId(id)
+            .setName(NAME)
+            .setCity(CITY)
+            .setIsMainCampus(FLAG_MAIN_CAMPUS)
+            .setLatitude(LATITUDE)
+            .setLongitude(LONGITUDE)
+            .setCarnegieSizeDegree(CarnegieSizeDegree.getDegree(CARNEGIE_SIZE_DEGREE))
+            .setAdmissionRate(ADMISSION_RATE)
+            .setAvgSat(AVG_SAT)
+            .setNumOfUndergrads(NUM_OF_UNDERGRADS)
+            .setAvgCost(AVG_COST)
+            .setRatioOfMen(RATIO_OF_MEN)
+            .setRatioOfWomen(RATIO_OF_WOMEN)
+            .build();
+    List<CollegeData> collegesData = Lists.newArrayList(college);
+    assertThat(collegesData).hasSize(1);
+
+    when(collegeDataApi.getCollegesById(ids)).thenReturn(collegesData);
+    List<MapsData> mapsData = mapsImpl.getMapsData(ids);
+
+    MapsData collegeMapData = mapsData.get(0);
+    assertThat(collegeMapData.name()).isEqualTo("New York University");
+    assertThat(collegeMapData.latitude()).isEqualTo(40.729452);
+  }
+
+  @Test
+  public void testEmptyMapsResourcesImpl() {
+    List<CollegeId> ids = new ArrayList<>();
+    List<CollegeData> collegesData = new ArrayList<>();
+
+    when(collegeDataApi.getCollegesById(ids)).thenReturn(collegesData);
+    List<MapsData> mapsData = mapsImpl.getMapsData(ids);
+    assertThat(mapsData).isEmpty();
+  }
+
+  @Test
+  public void testManyMapsResourcesImpl() {
+    CollegeId id1 = CollegeId.create(COLLEGE_ID);
+    CollegeId id2 = CollegeId.create(243744);
+    List<CollegeId> ids = Lists.newArrayList(id1, id2);
+    assertThat(ids).hasSize(2);
+
+    CollegeData college1 =
+        CollegeData.builder()
+            .setId(id1)
+            .setName(NAME)
+            .setCity(CITY)
+            .setIsMainCampus(FLAG_MAIN_CAMPUS)
+            .setLatitude(LATITUDE)
+            .setLongitude(LONGITUDE)
+            .setCarnegieSizeDegree(CarnegieSizeDegree.getDegree(CARNEGIE_SIZE_DEGREE))
+            .setAdmissionRate(ADMISSION_RATE)
+            .setAvgSat(AVG_SAT)
+            .setNumOfUndergrads(NUM_OF_UNDERGRADS)
+            .setAvgCost(AVG_COST)
+            .setRatioOfMen(RATIO_OF_MEN)
+            .setRatioOfWomen(RATIO_OF_WOMEN)
+            .build();
+    CollegeData college2 =
+        CollegeData.builder()
+            .setId(id2)
+            .setName("Stanford University")
+            .setCity("Stanford")
+            .setIsMainCampus(true)
+            .setLatitude(37.429434)
+            .setLongitude(-122.167359)
+            .setCarnegieSizeDegree(CarnegieSizeDegree.getDegree(CARNEGIE_SIZE_DEGREE))
+            .setAdmissionRate(0.0436)
+            .setAvgSat(1497.0)
+            .setNumOfUndergrads(7083)
+            .setAvgCost(69109)
+            .setRatioOfMen(0.4991)
+            .setRatioOfWomen(0.5009)
+            .build();
```

Comment:
For consistency with the NYU `CollegeData`, consider saving these fields as class member variables.

Reply:
+1

```java
  private static final CollegeData STANFORD_COLLEGE_DATA =
        CollegeData.builder()
            .setId(STANFORD_COLLEGE_ID)
            .setName("Stanford University")
            ...
            .build(); 
```